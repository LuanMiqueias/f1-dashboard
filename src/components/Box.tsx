import { ReactNode } from "react";
import styles from "../styles/components/Box.module.css";

interface Props {
  title: string;
  description?: string;
  button?: ReactNode;
}

const Box: React.FC<Props> = ({ children, title, button, description }) => {
  return (
    <div className={styles.box}>
      <div className={styles.content_top}>
        <div className={styles.title_box}>
          <h1>{title}</h1>
          {description && <p>{description}</p>}
        </div>
        {button}
      </div>
      {children}
    </div>
  );
};

export default Box;
