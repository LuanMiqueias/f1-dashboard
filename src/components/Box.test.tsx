import Box from "./Box";
import { render, screen } from "@testing-library/react";

it("Should math snapshot", () => {
  const { container } = render(<Box title="title 1" />);
  expect(container.firstChild).toMatchSnapshot();
});

it("Should have title", () => {
  render(<Box title="title 1" />);
  const title = screen.getByRole("heading", { name: /title 1/i });

  expect(title).toHaveTextContent(/title 1/i);
});
it("Should have description", () => {
  render(<Box title="title 1" description="description1" />);

  const description = screen.getByText(/description1/i);
  expect(description).toHaveTextContent(/description1/i);
});

it("Should have button", () => {
  const htmlBtn = <button>button1</button>;
  render(<Box title="title 1" button={htmlBtn} />);

  const button = screen.getByRole("button", { name: /button1/i });
  expect(button).toBeInTheDocument();
});
