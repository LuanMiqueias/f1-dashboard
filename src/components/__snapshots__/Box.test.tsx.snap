// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should math snapshot 1`] = `
<div
  class="box"
>
  <div
    class="content_top"
  >
    <div
      class="title_box"
    >
      <h1>
        title 1
      </h1>
    </div>
  </div>
</div>
`;
